### PR TITLE
Document missing start parameters and add shorthands for others.

### PR DESCRIPTION
**Pull request by TWiStErRob**
No issue number, just a PR.

### Context
I was looking for this magic https://twitter.com/ZacSweers/status/1453647736914399235. The first thing I checked was of course https://docs.gradle.org/current/userguide/command_line_interface.html and I couldn't find it there. Luckily I know a bit about the Gradle source code so I checked around there too and found it.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation
- [x] Recognize contributor in release notes

### CODE REVIEW

**Review by octylFractal — APPROVED** [review]

**Review by eskatos — CHANGES_REQUESTED** [review]
Good catch of missing options, thanks for your contribution! I requested some changes to make the documentation of the `--configuration-cache*` options more uniform with the other ones.

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 1]
```diff
@@ -339,6 +339,12 @@ Many of these options can be specified in `gradle.properties` so command-line fl
 `--build-cache`, `--no-build-cache`::
 Toggles the <<build_cache.adoc#build_cache, Gradle build cache>>. Gradle will try to reuse outputs from previous builds. _Default is off_.
 
+`--configuration-cache`, `--no-configuration-cache`::
+Toggles the <<configuration_cache.adoc#sec:config_cache,Configuration Cache>>. Gradle will try to reuse the build configuration from previous builds. This is an incubating and highly experimental feature, it is disabled by default.
```
[the remark this answers is not in the document]
Comment (TWiStErRob) [reply subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc:343]: Build cache is stable, config cache is not, shouldn't that be noted? Just by reading this page, people could think it's a first class, fully working and supported feature.

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 2]
```diff
@@ -339,6 +339,12 @@ Many of these options can be specified in `gradle.properties` so command-line fl
...
+
+`--configuration-cache-problems=(fail,warn)`::
+Configures how the configuration cache handles problems. Defaults to `fail`.
```
[the remark this answers is not in the document]
Comment (TWiStErRob) [reply subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc]: Oh, no, this Signoff thing doesn't let me merge suggestions from GitHub UI, 😔

**Review by eskatos — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (TWiStErRob) [reply subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc]: Sorry about that I was in bed last few days. Thanks for changing.

**Comment by TWiStErRob** [comment]
Is there anything else to do here?

**Comment by eskatos** [comment]
@TWiStErRob it's good to do I would say!

@bot-gradle test and merge

**Comment by eskatos** [comment]
@TWiStErRob some links are dead, could you please take a look?
https://ge.gradle.org/s/tseorngntonos/console-log?task=:docs:checkDeadInternalLinks

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 3]
```diff
@@ -339,6 +339,16 @@ Many of these options can be specified in `gradle.properties` so command-line fl
 `--build-cache`, `--no-build-cache`::
 Toggles the <<build_cache.adoc#build_cache, Gradle build cache>>. Gradle will try to reuse outputs from previous builds. _Default is off_.
 
+`--configuration-cache`, `--no-configuration-cache`::
+Toggles the <<configuration_cache.adoc#sec:config_cache,Configuration Cache>>. Gradle will try to reuse the build configuration from previous builds. _Default is off_.
```
Comment (TWiStErRob) [inline subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc:343]: ah, @eskatos I was trying to link to the top of the page. In adoc it has `[[config_cache]]`, how do I just link to the page without linking a section?
```suggestion
Toggles the <<configuration_cache.adoc,Configuration Cache>>. Gradle will try to reuse the build configuration from previous builds. _Default is off_.
```
?

Comment (TWiStErRob) [reply subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc:343]: Figured it out, ran this locally a few times
```
gradlew :docs:checkDeadInternalLinks
```
and experimented, the problem was that I thought the syntax is `#sec:somesection`, but it's just `#somesection` and `sec:` is a Gradle convention in some files to signify a section.

Comment (TWiStErRob) [reply subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc]: https://github.com/gradle/gradle/pull/18955/commits/5b1e6416698bf76ec8d2d1f248b3c078023e4b21

**Comment by TWiStErRob** [comment]
Should be good now, GHA CI green. DCO is green. And the failing task is passing locally (see https://github.com/gradle/gradle/pull/18955#discussion_r759568167).

I rebased on `master` too because the my last push failed with dependency resolution (see https://github.com/gradle/gradle/actions?query=branch%3Astartparams), might not be related but just in case.

Force-pushed a few times, hope that's fine considering Gradle core commits in the branch.

**Comment by octylFractal** [comment]
@bot-gradle test and merge

**Pull request merged**